src/smt/check_models.cpp

```diff
@@ -135,19 +167,48 @@ void CheckModels::checkModel(TheoryModel* m,
                << std::endl;
     std::stringstream ss;
     ss << "SolverEngine::checkModel(): "
-       << "ERRORS SATISFYING ASSERTIONS WITH MODEL:" << std::endl
-       << "assertion:     " << assertion << std::endl
-       << "simplifies to: " << n << std::endl
-       << "expected `true'." << std::endl
-       << "Run with `--check-models -v' for additional diagnostics.";
+       << "ERRORS SATISFYING ASSERTIONS WITH MODEL";
+    std::stringstream ssdet;
+    ssdet << ":" << std::endl
+          << "assertion:     " << assertion << std::endl
+          << "simplifies to: " << nval << std::endl
+          << "expected `true'." << std::endl
+          << "Run with `--check-models -v' for additional diagnostics.";
     if (hardFailure)
     {
+      // compute the theories involved, e.g. for the sake of issue tracking.
+      // to ensure minimality, if this is a topmost AND, miniscope
+      Node nmin = n;
+      while (nmin.getKind() == Kind::AND)
+      {
+        for (const Node& nc : nmin)
+        {
+          if (m->getValue(nc) == nval)
+          {
+            nmin = nc;
+            break;
+          }
+        }
+      }
+      // collect the theories of the assertion
+      std::vector<TheoryId> theories;
```

Comment:
One minor thing, this could be an unordered_set, then the `find` call is more efficient, but since this is debug code it's probably not too critical.